### PR TITLE
[test] convert bootstrap test configs to v2

### PR DESCRIPTION
**Pull request by derekargueta**
Description: Converts test configs for Bootstrap messages to v2. This also deletes code that depended on v1 Bootstrap translation. Bootstrap is a pretty fundamental config type, so this opens the door to fully cleaning out various other v1-v2 translation functions that were called by `translateBootstrap` such as `translateAddress` and `translateCluster`. For issue #6362.
Risk Level: Low
Testing: Included
Docs Changes: N/A
Release Notes: N/A

### CODE REVIEW

**Comment by derekargueta** [comment]
hmm, not sure about the doc failure since this doesn't touch docs, there's 1 warning in the output:
```
checking consistency... /source/generated/rst/configuration/best_practices/edge.rst: WARNING: document isn't included in any toctree
done
```

File: test/common/upstream/cluster_manager_impl_test.cc [hunk 1]
```diff
@@ -490,57 +505,57 @@ TEST_F(ClusterManagerImplTest, ValidClusterName) {
...
-    }]
-  }
+  const std::string yaml = R"EOF(
+static_resources:
+  clusters:
+  - name: cluster_1
+    connect_timeout: 0.250s
+    type: static
+    lb_policy: round_robin
+    hosts:
```
Comment (dio) [inline test/common/upstream/cluster_manager_impl_test.cc]: Should we directly convert `hosts` to `load_assignment` here? (since `hosts` is deprecated).

Comment (derekargueta) [reply test/common/upstream/cluster_manager_impl_test.cc]: Done. I did this for _most_ cases. There's a few spots where v1-consuming functions are still used (i.e. `configDump` used in `ClusterManagerImplTest.InitializeOrder`) but I'd prefer to handle those in a follow-up PR given the LoC already in this PR.

**Comment by derekargueta** [comment]
Thanks @dio for the doc fix!

**Pull request merged**